RFS/src/test/java/com/rfs/worker/SnapshotStepTest.java

```diff
@@ -0,0 +1,226 @@
+        return Stream.of(
+            Arguments.of(null, SnapshotStep.CreateEntry.class),
+            Arguments.of(new Snapshot("test", SnapshotStatus.NOT_STARTED), SnapshotStep.InitiateSnapshot.class),
+            Arguments.of(new Snapshot("test", SnapshotStatus.IN_PROGRESS), SnapshotStep.WaitForSnapshot.class)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideEnterPhaseNextArgs")
+    void EnterPhase_nextState_AsExpected(Snapshot snapshotEntry, Class<?> expected) {
+        // Set up the test
+        GlobalState globalState = Mockito.mock(GlobalState.class);
+        CmsClient cmsClient = Mockito.mock(CmsClient.class);
+        SnapshotCreator snapshotCreator = Mockito.mock(SnapshotCreator.class);
+
+        // Run the test
+        SnapshotStep.EnterPhase enterPhase = new SnapshotStep.EnterPhase(globalState, cmsClient, snapshotCreator, snapshotEntry);
+        WorkerStep nextState = enterPhase.nextStep();
+
+        // Check the results
+        assertEquals(expected, nextState.getClass());
+    }
+
+    @Test
+    void CreateEntry_run_AsExpected() {
+        // Set up the test
+        GlobalState globalState = Mockito.mock(GlobalState.class);
+        CmsClient cmsClient = Mockito.mock(CmsClient.class);
+        SnapshotCreator snapshotCreator = Mockito.mock(SnapshotCreator.class);
+        when(snapshotCreator.getSnapshotName()).thenReturn("test");
+
+        // Run the test
+        SnapshotStep.CreateEntry createPhase = new SnapshotStep.CreateEntry(globalState, cmsClient, snapshotCreator);
+        createPhase.run();
+
+        // Check the results
+        Mockito.verify(cmsClient, times(1)).createSnapshotEntry("test");
+    }
+
+    @Test
+    void CreateEntry_nextState_AsExpected() {
+        // Set up the test
+        GlobalState globalState = Mockito.mock(GlobalState.class);
+        CmsClient cmsClient = Mockito.mock(CmsClient.class);
+        SnapshotCreator snapshotCreator = Mockito.mock(SnapshotCreator.class);
+
+        // Run the test
+        SnapshotStep.CreateEntry createPhase = new SnapshotStep.CreateEntry(globalState, cmsClient, snapshotCreator);
+        WorkerStep nextState = createPhase.nextStep();
+
+        // Check the results
+        assertEquals(SnapshotStep.InitiateSnapshot.class, nextState.getClass());
+    }
+
+    @Test
+    void InitiateSnapshot_run_AsExpected() {
+        // Set up the test
+        GlobalState globalState = Mockito.mock(GlobalState.class);
+        CmsClient cmsClient = Mockito.mock(CmsClient.class);
+        SnapshotCreator snapshotCreator = Mockito.mock(SnapshotCreator.class);
+        when(snapshotCreator.getSnapshotName()).thenReturn("test");
+
+        // Run the test
+        SnapshotStep.InitiateSnapshot initiatePhase = new SnapshotStep.InitiateSnapshot(globalState, cmsClient, snapshotCreator);
+        initiatePhase.run();
+
+        // Check the results
+        Mockito.verify(snapshotCreator, times(1)).registerRepo();
```

Comment:
Taking a step back, are these tests useful?  The risk this is broken in these methods with little branches seems low as the collections of classes seem mostly declaritive.

I think it would be useful to be able to test the larger logic of the class, if I start migrating the snapshot, then issue X happens, do I end with the expected updates?